### ISSUE TITLE
Pytaco API: Bad operand type for unary -: 'tensor'

### ISSUE THREAD

**Issue by ByzanTine** [tensor-compiler/taco#342]
I am simply doing

```
>>> import pytaco as pt
>>> a = pt.tensor()
>>> -a
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: bad operand type for unary -: 'tensor'
```

Any hassle on supporting the unary operator?

**Comment by Infinoid** [comment]
What result would you expect?

The unary operator is defined on index expressions.

Example 1:
```python3
t = pytaco.tensor()
print(-t[None])
```

Output:
```
IndexExpr(-A0)
```

Example 2:
```python3
t = pytaco.tensor([10])
print(-t[5])
```

Output:
```
-0.0
```

**Comment by ByzanTine** [comment]
Thanks for the reply, I was expecting all values in the tensor to be negated, e.g like numpy API, where 
```
>>> import numpy
>>> import numpy as np
>>> np.ones(5)
array([ 1.,  1.,  1.,  1.,  1.])
>>> -np.ones(5)
array([-1., -1., -1., -1., -1.])
```

**Comment by Infinoid** [comment]
You can do that with index notation.  Taco is much more explicit about indexes than numpy is.

```python3
#!/usr/bin/env python3
from pytaco import tensor, index_var, from_array, Dense
import numpy

input = numpy.arange(20).reshape([5, 4])
x, y = [ index_var(c) for c in 'xy' ]

data = from_array(input)
negated = tensor([5, 4], [Dense, Dense])
print("data:", data)
negated[x, y] = -data[x, y]
print("negated:", negated)
```

Output:
```
data: A0 (5x4) ({dense},{dense}; 0,1):
dense (0): 
  [5]
dense (1): 
  [4]
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
negated: A6 (5x4) ({dense},{dense}; 0,1):
dense (0): 
  [5]
dense (1): 
  [4]
[0, -1, -2, -3, -4, -5, -6, -7, -8, -9, -10, -11, -12, -13, -14, -15, -16, -17, -18, -19]
```

**Comment by ByzanTine** [comment]
Ofc, index notation works, but from a user perspective, dealing a negation directly with indices sounds cumbersome?
FWIW, currently, in our code, we were hacking like:

`setattr(pt.tensor, '__neg__', lambda x: 0 - x)
`

**Comment by Infinoid** [comment]
Yeah, I see what you mean.

We've been talking about adding tensor-level operations, so that the syntax would look more like linear algebra... you could add or multiply vectors or matrices together without worrying about the indexes.  Unary negation of a tensor would be something we add at the same time.

**Comment by RawnH** [comment]
Ah I don't think I overloaded negation in PyTaco. I can take a look at adding that since I think it should be quick.

Some operand in PyTaco emulate numpy behavior already by creating index notation for the user. I can add one for negation as well.

**Comment by RawnH** [comment]
This is addressed in #350.

You can now do:

```python
#!/usr/bin/env python3
from pytaco import tensor, index_var, from_array, Dense
import numpy as np

arr_np = np.arange(20).reshape([5, 4])

data = from_array(arr_np)
print("data:", data)
print("negated:", -data)
```

To get the following output:
```
data: A0 (5x4) ({dense},{dense}; 0,1):
dense (0): 
  [5]
dense (1): 
  [4]
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
negated: A6 (5x4) ({dense},{dense}; 0,1):
dense (0): 
  [5]
dense (1): 
  [4]
[0, -1, -2, -3, -4, -5, -6, -7, -8, -9, -10, -11, -12, -13, -14, -15, -16, -17, -18, -19]
```

**Comment by ByzanTine** [comment]
Thanks for the fix! Verified it works.
